Demo/src/main/java/com/braintreepayments/demo/ShopperInsightsFragment.kt

```diff
@@ -31,142 +53,224 @@ class ShopperInsightsFragment : BaseFragment() {
+                        Toast.makeText(
+                            requireContext(),
+                            "Failed to authenticate user.",
+                            Toast.LENGTH_LONG
+                        )
+                            .show()
+                    }
+                }
+
+                is VenmoPendingRequest.Failure -> {}
+            }
+        }
+    }
+
+    private fun fetchShopperInsights() {
+        val email =
+            if (emailNullSwitch.isChecked) null else emailInput.editText?.text.toString()
+        val countryCode =
+            if (phoneNullSwitch.isChecked) null else countryCodeInput.editText?.text.toString()
+        val nationalNumber =
+            if (phoneNullSwitch.isChecked) null else nationalNumberInput.editText?.text.toString()
+
+        val request = if (countryCode != null && nationalNumber != null) {
+            ShopperInsightsRequest(email, ShopperInsightsBuyerPhone(countryCode, nationalNumber))
+        } else {
+            ShopperInsightsRequest(email, null)
+        }
+
+        shopperInsightsClient.getRecommendedPaymentMethods(
+            request
+        ) { result ->
+            when (result) {
+                is ShopperInsightsResult.Success -> {
+                    if (result.response.isPayPalRecommended) {
+                        payPalVaultButton.isEnabled = true
+                        shopperInsightsClient.sendPayPalPresentedEvent()
+                    }
+
+                    if (result.response.isVenmoRecommended) {
+                        venmoButton.isEnabled = true
+                        shopperInsightsClient.sendVenmoPresentedEvent()
+                    }
+
+                    responseTextView.text =
+                        """
+                            Eligible in PayPal Network: ${result.response.isEligibleInPayPalNetwork}
+                            PayPal Recommended: ${result.response.isPayPalRecommended}
+                            Venmo Recommended: ${result.response.isVenmoRecommended}
+                        """.trimIndent()
+                }
+
+                is ShopperInsightsResult.Failure -> {
+                    responseTextView.text = result.error.toString()
+                }
+            }
+        }
+    }
+
+    private fun launchPayPalVault() {
+        shopperInsightsClient.sendPayPalSelectedEvent()
+
+        payPalClient.createPaymentAuthRequest(
+            requireContext(),
+            PayPalRequestFactory.createPayPalVaultRequest(
+                activity,
+                emailInput.editText?.text.toString()
+            )
+        ) {
+            if (it == null) return@createPaymentAuthRequest
+            when (it) {
+                is PayPalPaymentAuthRequest.Failure -> {
+                    handleError(it.error)
+                }
+
+                is PayPalPaymentAuthRequest.ReadyToLaunch -> {
+                    paypalPendingRequest = paypalLauncher.launch(requireActivity(), it)
```

Comment:
I think we'd want to handle the `Started` or `Failure` pending request here - if it's `Started`, store it, if it's a failure display the error message. Same goes for Venmo, then we don't need to handle error scenarios in `handleReturnToApp` at that point it should only be a successfully started request